spinn_front_end_common/utility_models/data_speed_up_packet_gatherer_machine_vertex.py

```diff
@@ -272,22 +278,58 @@ def get_local_provenance_data(self):
                                     n_lost_seq_nums))))
         return prov_items
 
-    @staticmethod
     def set_cores_for_data_extraction(
-            transceiver, extra_monitor_cores_for_router_timeout,
+            self, transceiver, extra_monitor_cores_for_router_timeout,
             placements):
+
+        # Store the last reinjection status for resetting
+        # NOTE: This assumes the status is the same on all cores
+        self._last_reinjection_status = \
+            extra_monitor_cores_for_router_timeout[0].get_reinjection_status(
+                placements, transceiver)
+
         # set time out
         extra_monitor_cores_for_router_timeout[0].set_router_time_outs(
             15, 15, transceiver, placements,
             extra_monitor_cores_for_router_timeout)
+        extra_monitor_cores_for_router_timeout[0].\
+            set_reinjection_router_emergency_timeout(
+                1, 1, transceiver, placements,
+                extra_monitor_cores_for_router_timeout)
 
-    @staticmethod
     def unset_cores_for_data_extraction(
-            transceiver, extra_monitor_cores_for_router_timeout,
+            self, transceiver, extra_monitor_cores_for_router_timeout,
             placements):
-        extra_monitor_cores_for_router_timeout[0].set_router_time_outs(
-            15, 4, transceiver, placements,
-            extra_monitor_cores_for_router_timeout)
+        if self._last_reinjection_status is None:
+            logger.warn(
+                "Cores have not been set for data extraction, so can't be"
+                " unset")
+        try:
+            mantissa, exponent = \
+                self._last_reinjection_status.router_timeout_parameters()
+            extra_monitor_cores_for_router_timeout[0].set_router_time_outs(
+                mantissa, exponent, transceiver, placements,
+                extra_monitor_cores_for_router_timeout)
+            mantissa, exponent = \
+                self._last_reinjection_status\
+                .router_emergency_timeout_parameters()
+            extra_monitor_cores_for_router_timeout[0].\
+                set_reinjection_router_emergency_timeout(
+                    mantissa, exponent, transceiver, placements,
+                    extra_monitor_cores_for_router_timeout)
+        except Exception:
+            logger.error("Error resetting timeouts", exc_info=True)
+            logger.error("Checking if the cores are OK...")
+            core_subsets = convert_vertices_to_core_subset(
+                extra_monitor_cores_for_router_timeout, placements)
+            try:
+                error_cores = transceiver.get_cores_not_in_state(
+                    core_subsets, {CPUState.RUNNING})
+                if len(error_cores) > 0:
```

Comment:
This probably ought to be `if error_cores:`, since it's known to be a Python list.